aries-site/src/pages/components/textinput.js

```diff
@@ -17,21 +26,144 @@ const TextInput = () => (
+          behavior is important to screen reader users who need to know what
+          context the TextInput is referring to.
+        </SubsectionText>
+        <SubsectionText>
+          If you need to use TextInput outside of the context of a FormField, it
+          is important to make sure the TextInput is labeled in an alternate way
+          to meet accessibility requirements. One approach is to use another
+          visual indicator, such as the TextInput's icon, to serve as the label.
+          See how this is done in the{' '}
+          <Anchor href="#labeled-by-icon">Labeled by icon</Anchor> example.
+        </SubsectionText>
+        <SubsectionText>
+          Placeholder text does not serve as a sufficient means of meeting
+          accessibility requirements for labels. To meet accessbility
+          requirements, placeholder text should be used in conjunction with a
+          label or aria-labelledby attribute.
+        </SubsectionText>
+      </Subsection>
+    </ContentSection>
+    <ContentSection>
+      <Subsection name="Variants">
+        <SubsectionText>
+          A TextInput's visual state informs the user of its ability to be
+          interacted with or if any validation errors have occured.
+        </SubsectionText>
+      </Subsection>
+      <Subsection name="Password" level={3}>
+        <SubsectionText>
+          Used to maintain the privacy of the password a user has entered. This
+          is achieved by applying type="password" to the TextInput.
+        </SubsectionText>
+        <Example
+          docs="https://v2.grommet.io/button?theme=hpe#props"
+          code="https://raw.githubusercontent.com/hpe-design/design-system/master/aries-site/src/examples/components/textinput/TextInputPasswordExample.js"
+          height={{ min: 'small' }}
+        >
+          <TextInputPasswordExample />
+        </Example>
+      </Subsection>
+      <Subsection name="With suggestions" level={3}>
+        <SubsectionText>
+          Used to provide the user with available options or suggestions. These
+          suggestions can persist regardless of what the user has entered or can
+          filter based on what the user has typed.
+        </SubsectionText>
+        <Example
+          docs="https://v2.grommet.io/button?theme=hpe#props"
+          code="https://raw.githubusercontent.com/hpe-design/design-system/master/aries-site/src/examples/components/textinput/TextInputSuggestionsExample.js"
+          height={{ min: 'small' }}
+        >
+          <TextInputSuggestionsExample />
+        </Example>
+      </Subsection>
+      <Subsection name="Labeled by icon" level={3}>
+        <SubsectionText>
+          There may be some cases where you need to use a TextInput without a
+          visual label. To meet accessibility requirements, the TextInput can be
+          labeled by another visual context such as an icon with an id.
```

Comment:
Voice: trying to avoid use of 'you' for the time being -- unless we adopt a more casual voice -- tbd.

```suggestion
          Some situations may need TextInput without a visual label, 
          such as when implementing a search field. To meet 
          accessibility requirements, the TextInput may be labeled by 
          another visual context such as an icon with an id.
```